gnat2goto/driver/tree_walk.adb

```diff
@@ -2190,25 +2235,62 @@ package body Tree_Walk is
                                       "Original node not present");
       end if;
       N_Orig := Original_Node (N);
-      if Pragma_Name (N_Orig) in Name_Assert | Name_Assume |
-         Name_Loop_Invariant
-      then
-         Do_Pragma_Assert_or_Assume (N_Orig, Block);
-      --  Ignore here. Rather look for those when we process a node.
-      elsif Pragma_Name (N_Orig) in Name_Annotate then
-         null;
-      --  The following pragmas are currently unimplemented, we ignore them
-      --  here
-      elsif Pragma_Name (N_Orig) in Name_SPARK_Mode | Name_Global |
-         Name_Postcondition | Name_Refined_State | Name_Refined_Global |
-         Name_Precondition
-      then
-         Report_Unhandled_Node_Empty (N, "Do_Pragma_Assert_or_Assume",
-                                      "Unsupported pragma");
-      else
-         Report_Unhandled_Node_Empty (N, "Do_Pragma_Assert_or_Assume",
-                                      "Unknown");
-      end if;
+      case Pragma_Name (N_Orig) is
+         when Name_Assert | Name_Assume | Name_Loop_Invariant =>
+            Do_Pragma_Assert_or_Assume (N_Orig, Block);
+         when Name_Annotate =>
+            --  Ignore here. Rather look for those when we process a node.
+            null;
+         when Name_SPARK_Mode | Name_Global | Name_Postcondition |
+              Name_Refined_State | Name_Refined_Global | Name_Precondition =>
+            --  These pragmas are currently unimplemented
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                         "Unsupported pragma");
+         when Name_All_Calls_Remote |
+              Name_Asynchronous |
+              Name_Atomic |
+              Name_Atomic_Components |
+              Name_Attach_Handler |
+              Name_Controlled |
+              Name_Convention |
+              Name_Discard_Names |
+              Name_Elaborate |
+              Name_Elaborate_All |
+              Name_Elaborate_Body |
+              Name_Export |
+              Name_Import |
+              Name_Inline |
+              Name_Inspection_Point |
+              Name_Interrupt_Handler |
+              Name_Linker_Options |
+              Name_List |
+              Name_Locking_Policy |
+              Name_Normalize_Scalars |
+              Name_Optimize |
+              Name_Pack |
+              Name_Page |
+              Name_Preelaborate |
+              Name_Pure |
+              Name_Queuing_Policy |
+              Name_Remote_Call_Interface |
+              Name_Remote_Types |
+              Name_Restrictions |
+              Name_Reviewable |
+              Name_Shared_Passive |
+              Name_Storage_Size |
+              Name_Task_Dispatching_Policy |
+              Name_Unsuppress |
+              Name_Volatile |
+              Name_Volatile_Components =>
+            --  These pragmas are currently ignored
```

Comment:
I would need to go through the Appendix but I don't think it is safe to ignore all of these.

Reply:
At the very least the elaboration related pragmas can have an impact on the semantics of an Ada program.